transport/register.go

```diff
@@ -92,33 +95,43 @@ func (m MapRegistry) Register(rs []Registrant) {
 			r.Service = m.defaultService
 		}
 
-		m.unaryEntries[ServiceProcedure{r.Service, r.Procedure}] = r.Handler
+		switch r.HandlerSpec.Type {
+		case Unary:
+			m.entries[ServiceProcedure{r.Service, r.Procedure}] = r.HandlerSpec.UnaryHandler
+		case Oneway:
+			m.onewayEntries[ServiceProcedure{r.Service, r.Procedure}] = r.HandlerSpec.OnewayHandler
+		default:
+			panic(fmt.Sprintf("Unknown RPC Type %v, for %s::%s", r.HandlerSpec.Type, r.Service, r.Procedure))
```

Comment:
The `::` pattern is for thrift functions. Explicitly say service and procedure here.

> Unknown RPC type %v for service %q, procedure %q